test/mitmproxy/addons/test_proxyauth.py

```diff
@@ -109,6 +121,24 @@ def test_check():
         )
         assert not up.check(f)
 
+        ctx.configure(
```

Comment:
Ok, to sum up : 
If we are not testing the connection itself, we can use a mock ldap server to test that we do search-write-update etc... correctly
If we are testing the LDAP connection then we are really doing an integration test. In which case it's probably best to connect to a real LDAP implementation.
In our case, we are just establishing a connection to an ldap server and performing one search request, so for the connection part I don't think there is a viable solution for off line testing (apart from running a full -even minimal- local ldap server). I could still take the search part out of the function (check) and do a unit test for that part by mocking an ldap server.
What do you think ?